.github/workflows/ephemeral-env.yml

```
@@ -27,8 +27,28 @@ jobs:
     outputs:
       slash-command: ${{ steps.eval-body.outputs.result }}
       feature-flags: ${{ steps.eval-feature-flags.outputs.result }}
+      sha: ${{ steps.get-sha.outputs.sha }}
 
     steps:
+    - name: Get Info from comment
+      uses: actions/github-script@v7
+      id: get-pr-info
+      with:
+        script: |
+          const request = {
+              owner: context.repo.owner,
+              repo: context.repo.repo,
+              pull_number: ${{ github.event.inputs.issue_number || github.event.issue.number }},
+          };
+          core.info(`Getting PR #${request.pull_number} from ${request.owner}/${request.repo}`);
+          const pr = await github.rest.pulls.get(request);
+          return pr.data;
+
+    - name: Debug
+      id: get-sha
+      run: |
+        echo "sha=${{ fromJSON(steps.get-pr-info.outputs.result).head.sha }}" >> $GITHUB_OUTPUT
```

You should exit early if the PR was updated after the comment. See https://github.com/apache/camel/blob/66fbdcd2c71a6588bacd7b3e0d2a03128c0cd069/.github/workflows/pr-comment.yml#L55-L57

```suggestion
    - name: Debug
      id: get-sha
      env:
      	COMMENT_AT: ${{ github.event.comment.created_at }}
      	PUSHED_AT: ${{ fromJSON(steps.get-pr-info.outputs.result).pushed_at }}
      run: |
        if [[ $(date -d "$PUSHED_AT" +%s) -gt $(date -d "$COMMENT_AT" +%s) ]]; then
            exit 1
        fi
        echo "sha=${{ fromJSON(steps.get-pr-info.outputs.result).head.sha }}" >> $GITHUB_OUTPUT
```

---

Unfortunately, this is not useful as `pushed_at` field is deprecated and the value will be nil.

---

I don't see any evidence of that in the documentation. Can you point me to your source?

---

If untrusted code needs to be run, I would move the IssueOps workflow to a Label gate (pull_request + label). There is no way to get a trustful commit date.

---

just to tie things together: `pushed_at` from https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#get-a-pull-request is slightly different from `Commit.pushedDate`, but I suppose the they're ultimately populated from the same source, so both are deprecated/going away? This wasn't obvious to me before but would make sense. I didn't see anything in https://docs.github.com/en/rest/about-the-rest-api/breaking-changes but I guess it just somewhat follows from https://docs.github.com/en/graphql/overview/breaking-changes .

---

@pwntester @raboof sorry for the delay here. I've updated to workflow to use PR labels, would be great if you can review it again